.circleci/config.yml

```diff
@@ -85,22 +86,45 @@ jobs:
       - setup-vault
       - run: go test -count=1 -timeout=80s -tags 'integration vault' ./... -run Vault
 
+  split-e2e-tests:
+    executor:
+      name: go
+    resource_class: medium
+    steps:
+      - checkout
+      - run:
+          name: Split E2E tests
+          command: |
+            ./build-scripts/list-tests.sh e2e ./e2e | ./build-scripts/split-lines.sh ${E2E_TESTS_PARALLELISM} .build/tests
+      - persist_to_workspace:
+          root: .build
+          paths:
+            - tests
+
   e2e-tests:
     executor:
       name: go
-    resource_class: medium+
+    resource_class: medium
     parameters:
       part_index:
         type: integer
     steps:
       - checkout
       - setup-consul
       - setup-gotestsum
+      - attach_workspace:
+          at: .build
+      - setup-terraform
       - run:
           name: E2E tests
+          environment:
+            TESTS_LIST_PATH: .build/tests/part.<< parameters.part_index >>
+            TESTS_REGEX_PATH: .build/regex/part.<< parameters.part_index >>
           command: |
-            ./build-scripts/split-tests.sh e2e ./e2e ${E2E_TESTS_PARALLELISM} ${PWD}/.build/chunks
-            CHUNK_INDEX=<< parameters.part_index >> make test-e2e-circleci
+            mkdir -p .build/regex
+            sudo cp terraform /home/circleci/project/e2e/terraform
```

Comment:
Not having this line is why the e2e tests have been intermittently failing. Without it, multiple CTS instances (running in parallel) determine that TF is not installed and they try to install it and copy it at the same time. Resulting in "text file busy" error from fork/exec when a CTS instance tries to run TF when another CTS instance is writing (installing) to it.